### PR TITLE
[Alerting]: Add /api prefix to all routes

### PR DESCRIPTION
**Pull request by papagian**
**What this PR does / why we need it**:
Following [this](https://github.com/grafana/grafana/pull/32688#discussion_r607052032) discussion.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
Requires grafana/alerting-api#43
Target grafana/grafana#32688

The path is [re-written](https://github.com/grafana/grafana/blob/693985a6ba0b0584ab3b2a9090df61ee2debf93d/pkg/services/ngalert/api/lotex.go#L45) when we forward requests to loki/cortex/aletmanager so no need for extra handling.

### CODE REVIEW

File: pkg/services/ngalert/api/api_ruler_base.go [hunk 1]
```diff
@@ -33,12 +33,12 @@ type RulerApiBase struct {
 
 func (api *API) RegisterRulerApiEndpoints(srv RulerApiService) {
 	api.RouteRegister.Group("", func(group routing.RouteRegister) {
```
Comment (kylebrandt) [inline pkg/services/ngalert/api/api_ruler_base.go:35]: Is there some reason we can't make the the first argument to the `Group` method `"/api/ruler"` in this and other sections to shorten the repetition?

**Review by kylebrandt — APPROVED** [review]
Works for me :-) Also nicer now the something like `/api/rulerz/` will return a JSON 404 not found instead of some HTML +1

Added a note/question about using the Group prefix.

**Comment by gotjosh** [comment]
FYI @domasx2

**Review by gotjosh — APPROVED** [review]

**Review by domasx2 — APPROVED** [review]

File: pkg/services/ngalert/api/api_ruler_base.go (hunk 1 above)
Comment (papagian) [reply pkg/services/ngalert/api/api_ruler_base.go:35]: good point! This is auto-generate from [this](https://github.com/grafana/grafana/blob/664268a49803edfbd0cd25378e39ae6667901644/pkg/services/ngalert/api/swagger-codegen/templates/controller-api.mustache#L25) template so I hadn't found an easy way to make it work for all the routes.

Comment (kylebrandt) [reply pkg/services/ngalert/api/api_ruler_base.go:35]: Oh auto-gen, doesn't matter then :-)

**Pull request merged**